osrf_testing_tools_cpp/src/memory_tools/impl/linux.cpp

```diff
@@ -42,13 +42,21 @@ find_original_function(const char * name)
   return original_function;
 }
 
+// An amount of memory that is greater than what is needed for static initialization
+// for any test we run. It was found experimentally on Ubuntu Linux 16.04 x86_64.
+static constexpr size_t STATIC_ALLOCATOR_SIZE = 0x800000;
 using osrf_testing_tools_cpp::memory_tools::impl::StaticAllocator;
-// the size was found experimentally on Ubuntu Linux 16.04 x86_64
-using StaticAllocatorT = StaticAllocator<8388608>;
-// used to fullfil calloc call from dlerror.c during initialization of original functions
-// constructor is called on first use with a placement-new and the static storage
+using StaticAllocatorT = StaticAllocator<STATIC_ALLOCATOR_SIZE>;
 static uint8_t g_static_allocator_storage[sizeof(StaticAllocatorT)];
-static StaticAllocatorT * g_static_allocator = nullptr;
+
+// Contains global allocator to make 100% sure to avoid Static Initialization Order Fiasco.
+// "Construct on first use" idiom
+StaticAllocatorT * get_static_allocator() {
```

Comment:
This package cannot depend on ament_lint, but if it could I would do it :) To that end please format this with the `{` on a new line and for my preference the return type on its own line:

```c++
StaticAllocatorT *
get_static_allocator()
{
```

Also, please make it `static`, as no one else needs to see it outside of the TU.

Reply:
Why can't it? Does it introduce a circular dependency? Barring that, could we hook it up to a Travis job that runs linters non-intrusively?

Will address the formatting

Reply:
You could use `uncrustify --reformat` BTW even if you don't have the linter plugged into this package for this PR.

Reply:
I intentionally made this package pure cmake, rather than depend on ament so that some others could use it (it's only dependency is cmake).

We could have a travis job that installs and runs `ament_*` linters, but I just haven't done that.